java-frontend/src/main/java/org/sonar/java/se/ExplodedGraph.java

```diff
@@ -113,7 +117,7 @@ public Tree syntaxTree() {
     public final ProgramPoint programPoint;
     @Nullable
     public final ProgramState programState;
-    private final List<Node> parents;
+    private final Map<Node, MethodYield> parents;
```

Comment:
I would correct it by saying that:
> A node can only be reached from a parent via **at most** one method yield.
MethodYields are optionals.

Now, I didn't noticed any cases where it was not the case, because of uniqueness of yields, which always imply learning something different. I didn't enforced that property with a preconditions, but it maybe worth it. WDYT?